utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/kafka/KafkaConnector.java

```diff
@@ -85,17 +86,16 @@ public void produceMessageWithoutNewThread(String message) throws Exception {
     }
 
     // For APM testing, a consume message without starting a new thread
-    public boolean consumeMessageWithoutNewThread(Integer timeout_s) throws Exception {
-        KafkaConsumer<String, String> consumer = createKafkaConsumer();
+    public boolean consumeMessageWithoutNewThread(Integer timeout_s, String topic) throws Exception {
+        KafkaConsumer<String, String> consumer = createKafkaConsumer(topic);
         consumer.subscribe(Collections.singletonList(topic));
-        ConsumerRecords<String, String> records = consumer.poll(Duration.ofSeconds(timeout_s));
-        if(records.isEmpty()){
-            System.out.println("No record found when polling " + topic);
-            return false;
+        boolean recordFound = false;
+        while (true) {
+            ConsumerRecords<String, String> records = consumer.poll(Duration.ofMillis(100));
+            for (ConsumerRecord<String, String> record : records)
```

Comment:
I think you missed a bracket `{` here (to be closed after recordFound = true ? Otherwise only one instruction will be in the for loop

Reply:
This was the problem, thanks!